Makefile

```diff
@@ -113,3 +102,71 @@ generate-pip-lockfiles: auth/pinned-requirements.txt
 generate-pip-lockfiles: batch/pinned-requirements.txt
 generate-pip-lockfiles: ci/pinned-requirements.txt
 generate-pip-lockfiles: memory/pinned-requirements.txt
+
+$(HAILTOP_VERSION):
+	$(MAKE) -C hail python/hailtop/hail_version
+
+hail-ubuntu-image: docker/hail-ubuntu
+	$(eval HAIL_UBUNTU_IMAGE := $(DOCKER_PREFIX)/hail-ubuntu:$(TOKEN))
+	python3 ci/jinja2_render.py '{"global":{"docker_prefix":"$(DOCKER_PREFIX)"}}' docker/hail-ubuntu/Dockerfile docker/hail-ubuntu/Dockerfile.out
+	./docker-build.sh docker/hail-ubuntu Dockerfile.out $(HAIL_UBUNTU_IMAGE)
+	echo $(HAIL_UBUNTU_IMAGE) > $@
+
+base-image: hail-ubuntu-image docker/Dockerfile.base
+	$(eval BASE_IMAGE := $(DOCKER_PREFIX)/base:$(TOKEN))
+	python3 ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat hail-ubuntu-image)'"}}' docker/Dockerfile.base docker/Dockerfile.base.out
+	./docker-build.sh . docker/Dockerfile.base.out $(BASE_IMAGE)
+	echo $(BASE_IMAGE) > $@
+
+private-repo-hailgenetics-hail-image: hail-ubuntu-image docker/hailgenetics/hail/Dockerfile $(shell git ls-files hail/src/main hail/python)
+	$(eval PRIVATE_REPO_HAILGENETICS_HAIL_IMAGE := $(DOCKER_PREFIX)/hailgenetics/hail:$(TOKEN))
```

Comment:
this is so that you don't have to smoosh the docker-build.sh, rm, and echo into one bash invocation, yeah?